test/support/connection_test_helper.ex

```diff
@@ -0,0 +1,20 @@
+defmodule Janus.ConnectionTest.Helper do
+  alias Janus.Connection
+  alias FakeHandler.Payloads
+
+  def handlers_last_callback_call({:state, _, _, _, %{callback: callback}, _}), do: callback
+
+  # helper test macro to generate test for checking if proper Handler's callback has been called
+  defmacro test_callback(event) do
+    fun = String.to_atom("handle_" <> Atom.to_string(event))
+
+    quote do
+      test "#{inspect(unquote(fun))} callback", %{state: state} do
```

Comment:
This test expects state to be provided and I don't see any docs anywhere saying it does. This can lead to confusion if someone wants to use it in distant future.